### PR TITLE
Ability to set generic errorHandler for all reqs

### PR DESCRIPTION
**Pull request by iest**
Particularly useful for blanket responding to a 401.

### CODE REVIEW

File: test/PactAPI-test.js [hunk 1]
```diff
@@ -49,14 +56,23 @@ describe('PactAPI', () => {
     });
   });
 
+  describe('setErrorHandler', () => {
```
Comment (rdjpalmer) [inline test/PactAPI-test.js]: Seems like it would make sense to also test that the error handler is run on a failed request.

Comment (rdjpalmer) [inline test/PactAPI-test.js]: Apologies, didn't see below :)

Comment (iest) [inline test/PactAPI-test.js]: It 100% works, but testing it is gonna require us to change how we mock superagent. Currently it's a basic stub and doesn't really work very well.

Comment (rdjpalmer) [inline test/PactAPI-test.js]: Realised that after I left the comment!

**Comment by rdjpalmer** [comment]
:ship:

**Pull request merged**